diff -urN so3g-0.1.13_orig/setup.py so3g-0.1.13/setup.py
--- so3g-0.1.13_orig/setup.py	2024-03-29 08:40:04.000000000 -0700
+++ so3g-0.1.13/setup.py	2024-04-02 22:41:13.998376523 -0700
@@ -152,7 +152,12 @@
     sp.check_call(["cmake", cmake_list_dir] + cmake_args, cwd=build_dir, env=env)
 
     print("-" * 10, "Building {}".format(pkg), "-" * 40)
-    cmake_cmd = ["cmake", "--build", "."] + build_args + ["--", "-j2"]
+    if "CPU_COUNT" in os.environ:
+        # Running in conda-build
+        parallel_args = ["--parallel", os.environ["CPU_COUNT"]]
+    else:
+        parallel_args = ["--parallel", "2"]
+    cmake_cmd = ["cmake", "--build", "."] + build_args + parallel_args
     sp.check_call(cmake_cmd, cwd=build_dir)
     cmake_cmd = ["cmake", "--install", "."] + build_args
     sp.check_call(cmake_cmd, cwd=build_dir)
