diff -urN so3g-0.1.8_orig/setup.py so3g-0.1.8/setup.py
--- so3g-0.1.8_orig/setup.py	2023-08-17 17:04:38.000000000 -0700
+++ so3g-0.1.8/setup.py	2023-09-27 22:17:48.413193444 -0700
@@ -371,7 +371,6 @@
 conf["url"] = "https://github.com/simonsobs/so3g"
 conf["version"] = get_version()
 conf["python_requires"] = ">=3.7.0"
-conf["setup_requires"] = (["wheel", "cmake"],)
 conf["install_requires"] = [
     "numpy",
     "astropy",
@@ -381,7 +380,7 @@
     "pytz",
     "pyaml",
     "sqlalchemy",
-    "pysqlite3-wheels",
+    "pysqlite3",
     "tqdm",
     "qpoint",
 ]
