diff -urN moby2-f0056ac3618b8c4c96130fcb41f7be3d6bd07241_prog/src/dirfile.c moby2-f0056ac3618b8c4c96130fcb41f7be3d6bd07241/src/dirfile.c
--- moby2-f0056ac3618b8c4c96130fcb41f7be3d6bd07241_prog/src/dirfile.c	2023-01-13 09:34:41.000000000 -0800
+++ moby2-f0056ac3618b8c4c96130fcb41f7be3d6bd07241/src/dirfile.c	2025-02-26 20:05:51.407903220 -0800
@@ -40,9 +40,10 @@
 
 static PyTypeObject
 ptrobjType = {
-    PyObject_HEAD_INIT(NULL)
 #if PY_MAJOR_VERSION >= 3
+    PyVarObject_HEAD_INIT(NULL, 0)
 #else
+    PyObject_HEAD_INIT(NULL)
     0,                         /* ob_size */
 #endif
     "ptrobj",                  /* tp_name */
