diff -urN moby2-f0056ac3618b8c4c96130fcb41f7be3d6bd07241_prog/pyproject.toml moby2-f0056ac3618b8c4c96130fcb41f7be3d6bd07241/pyproject.toml
--- moby2-f0056ac3618b8c4c96130fcb41f7be3d6bd07241_prog/pyproject.toml	1969-12-31 16:00:00.000000000 -0800
+++ moby2-f0056ac3618b8c4c96130fcb41f7be3d6bd07241/pyproject.toml	2025-02-26 19:41:06.768744151 -0800
@@ -0,0 +1,41 @@
+[build-system]
+requires = [
+    "setuptools",
+    "wheel",
+    "numpy",
+    # Astropy depends on numpy 1.x with python-3.9.  Place
+    # a build-time dependency here so that we build with a
+    # compatible version of numpy.  Remove this after dropping
+    # python-3.9 support.
+    "astropy",
+]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "moby2"
+readme = "README.rst"
+description = "ACTpol support"
+urls = {source = "https://github.com/ACTCollaboration/moby2"}
+license = {file = "LICENSE"}
+requires-python = ">=3.9"
+dependencies = [
+    "future",
+    "numpy",
+    "scipy",
+    "astropy",
+    "matplotlib",
+    "ephem",
+    "pytz",
+]
+version = "0.1"
+classifiers = [
+  "Development Status :: 5 - Production/Stable",
+  "Environment :: Console",
+  "Intended Audience :: Science/Research",
+  "License :: OSI Approved :: BSD License",
+  "Topic :: Scientific/Engineering :: Astronomy",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+]
diff -urN moby2-f0056ac3618b8c4c96130fcb41f7be3d6bd07241_prog/setup.py moby2-f0056ac3618b8c4c96130fcb41f7be3d6bd07241/setup.py
--- moby2-f0056ac3618b8c4c96130fcb41f7be3d6bd07241_prog/setup.py	2023-01-13 09:34:41.000000000 -0800
+++ moby2-f0056ac3618b8c4c96130fcb41f7be3d6bd07241/setup.py	2025-02-26 19:38:34.289271999 -0800
@@ -1,8 +1,7 @@
 from __future__ import print_function
 
-from distutils.core import setup, Extension
+from setuptools import setup, Extension
 
-VERSION = '0.1'
 
 import glob
 import sys
@@ -44,26 +43,24 @@
 if sys.version_info.major >= 3:
     analysis_modules.append('socompat')
 
-setup (name = 'moby2',
-       version = VERSION,
-       description = 'ACTpol support',
-       package_dir = {'moby2': 'python'},
-       ext_modules = [module1],
-       scripts = scripts,
-       packages = (
-           ['moby2',
-            'moby2.aux_data',
-            'moby2.detectors',
-            'moby2.ephem',
-            'moby2.instruments',
-            'moby2.instruments.actpol',
-            'moby2.instruments.mbac',
-            'moby2.mapping',
-            'moby2.pointing',
-            'moby2.scripting',
-            'moby2.tod',
-            'moby2.util',
-            'moby2.analysis'] +
-           ['moby2.analysis.%s' % p for p in analysis_modules]
-       )
-   )
+setup(name = 'moby2',
+    package_dir = {'moby2': 'python'},
+    ext_modules = [module1],
+    scripts = scripts,
+    packages = (
+        ['moby2',
+        'moby2.aux_data',
+        'moby2.detectors',
+        'moby2.ephem',
+        'moby2.instruments',
+        'moby2.instruments.actpol',
+        'moby2.instruments.mbac',
+        'moby2.mapping',
+        'moby2.pointing',
+        'moby2.scripting',
+        'moby2.tod',
+        'moby2.util',
+        'moby2.analysis'] +
+        ['moby2.analysis.%s' % p for p in analysis_modules]
+    )
+)
